### PR TITLE
 stm32 adc define the internal voltage reference

### PR DESCRIPTION
**Pull request by FRASTM**
This change is to define the ref_internal as a pvalue of the adc_driver_api
This Analog reference voltage for the ADC is given by the board AVDD to the VREF+ input of the MCU
It is usually connected to the VDD supply line.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -811,6 +812,11 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
+
+	if (config->has_vbat_channel &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VBAT) == channel_id) {
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:817]: To respect API (cf similar change done in https://github.com/zephyrproject-rtos/zephyr/pull/43857)
```suggestion
		__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_VBAT) == channel_id) {
```

Comment (erwango) [inline drivers/adc/adc_stm32.c:817]: U5 also has LL_ADC_CHANNEL_VBAT and LL_ADC_CHANNEL_VBAT_ADC4 which looks like a different channel.

**Review by erwango — CHANGES_REQUESTED** [review]
Miss dts & bindings updates

File: drivers/adc/adc_stm32.c (hunk 1 above)
Comment (FRASTM) [reply drivers/adc/adc_stm32.c]: removing this part from that PR

**Comment by FRASTM** [comment]
change this PR to only have the **ref_internal** setting

File: dts/bindings/adc/st,stm32-adc.yaml [hunk 2]
```diff
@@ -27,6 +27,11 @@ properties:
     pinctrl-names:
       required: true
 
+    vref_mv:
+      type: int
+      default: 3300
+      description: Indicates the reference voltage of the ADC in mV.
```
Comment (erwango) [inline dts/bindings/adc/st,stm32-adc.yaml:33]: Required:false

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/bindings/adc/st,stm32-adc.yaml (hunk 2 above)
Comment (FRASTM) [reply dts/bindings/adc/st,stm32-adc.yaml:34]: adding
`       required: false`

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Comment by ycsin** [comment]
@FRASTM @erwango is this something that should be backported, if it fixes anything?

**Comment by erwango** [comment]
@ycsin Please don't add backport label on all PRs. As already explained, only bugfixes should be backported (and not all of them). This is not the case here. Please see discussions
https://github.com/zephyrproject-rtos/zephyr/discussions/43561 and  https://github.com/zephyrproject-rtos/zephyr/issues/43335

**Comment by ycsin** [comment]
@erwango I actually think that this PR would fit into "Usability Bug Fixes and improvements", since it will make the [adc sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/adc/src/main.c#L94) work.

**Comment by erwango** [comment]
> @erwango I actually think that this PR would fit into "Usability Bug Fixes and improvements", since it will make the [adc sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/adc/src/main.c#L94) work

So indeed, "Usability Bug Fixes and improvements" is the part which is being discussed and that some would like to prevent being merged. We should focus on keeping the branch stable as it should be maintained for next 2 years. So we should minimize the changes and put prio on actual fw **bugs**.

**Pull request merged**